src/esp/physics/bullet/BulletArticulatedObject.h

```diff
@@ -222,6 +231,18 @@ class BulletArticulatedObject : public ArticulatedObject {
   std::unique_ptr<btCompoundShape> bFixedObjectShape_;
   std::unique_ptr<btRigidBody> bFixedObjectRigidBody_;
 
+  // TODO: should be stored in the link
+  // compound parent collision shapes for the links
+  std::map<int, std::unique_ptr<btCompoundShape>> linkCompoundShapes_;
+
+  // child mesh convex and primitive shapes for the link compound shapes
+  std::map<int, std::vector<std::unique_ptr<btCollisionShape>>>
```

Comment:
Shouldn't this be a std::multimap instead? That will be a lot more efficient the managing all these std::vectors as values.

Reply:
You may be right - those were not a part of this PR, they were just moved around to organize things a little cleanly.